### PR TITLE
solution

### PR DESCRIPTION
**Pull request by 1stIndominus**

### CODE REVIEW

File: src/averageAges.js [hunk 1]
```diff
@@ -20,6 +20,15 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const men = people.filter(person => person.sex === 'm');
+
+  const menToFilter = century
+    ? men.filter(person => Math.ceil(person.died / 100) === century)
+    : men;
```
Comment (SanyaBratashchuk) [inline src/averageAges.js:27]: You can call filter once passing different callbacks

File: src/averageAges.js [hunk 2]
```diff
@@ -20,6 +20,15 @@ function calculateMenAverageAge(people, century) {
...
+
+  return menToFilter.reduce((sum, person, index, array) => {
+    return sum + (person.died - person.born) / array.length;
+  }, 0);
```
Comment (SanyaBratashchuk) [inline src/averageAges.js:31]: move it to the seperate function and reuse it in your main functions

File: src/averageAges.js [hunk 3]
```diff
@@ -38,6 +47,15 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  const menToFilter = withChildren
+    ? women.filter(person => people.some(child => child.mother === person.name))
+    : women;
```
Comment (SanyaBratashchuk) [inline src/averageAges.js:54]: The same pass different callbacks

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]

File: src/averageAges.js [hunk 4]
```diff
@@ -15,11 +15,18 @@
...
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(
+    century
+      ? person => {
+        const aprovCentury = Math.ceil(person.died / 100) === century;
```
Comment (SanyaBratashchuk) [inline src/averageAges.js:21]: aprovCentury? use more informative names

File: src/averageAges.js [hunk 5]
```diff
@@ -15,11 +15,18 @@
...
+
+        return aprovCentury && person.sex === 'm';
+      }
+      : person => person.sex === 'm');
+
+  return men.reduce((sum, person, index, array) => {
+    return sum + (person.died - person.born) / array.length;
+  }, 0);
```
Comment (SanyaBratashchuk) [inline src/averageAges.js:29]: move it to the separate function and reuse it in your main functions

File: src/averageAges.js [hunk 6]
```diff
@@ -55,7 +73,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const averagePeopleAge = onlyWithSon
```
Comment (SanyaBratashchuk) [inline src/averageAges.js:76]: it is not an averagePeopleAge

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]

File: src/averageAges.js [hunk 7]
```diff
@@ -55,7 +69,28 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const ageDifference = onlyWithSon
+    ? people.filter(child => child.sex === 'm')
+    : people;
```
Comment (Bohdan-Maksymiuk) [inline src/averageAges.js:74]:
```suggestion
  const ageDifference = onlyWithSon
    ? people.filter(child => child.sex === 'm')
    : people;
```
is it ageDifference?

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]

**Review by Bohdan-Maksymiuk — APPROVED** [review]

**Pull request closed without merging**